From 0372481ec66719a78787e08d41b16e4ae3a477ac Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Mon, 26 Feb 2018 20:28:40 +0100
Subject: [PATCH 2/2] Revert "sepolicy: Policy for recovery"

Revert "sepolicy: Isolate qcom sysfs_graphics policy for recovery"

This reverts commit c4c96f6145838adb949b204ea7e3828ec2eb72a7.

Revert "recovery: Clean up recovery SELinux policy"

This reverts commit 2c9cfd4cb815bba6cdb6418e5006069cb8f26d85.

Revert "common: Add sf_lcd_density_prop type and labelled props"

This reverts commit d54feea5d968f51d66421605e178be1f43015dd5.

Revert "sepolicy: Initial policy for recovery"

This reverts commit 5e84bfc96ec99eb17be5fe09e9034e83590bf1a6.

Change-Id: I7ebede8ee13c7a81e7954cc243b94c108a4dafad
---
 common/private/file_contexts     |  3 ---
 common/private/property.te       |  1 -
 common/private/property_contexts |  2 --
 common/private/recovery.te       | 35 -----------------------------------
 common/private/system_app.te     |  3 ---
 common/private/vold.te           |  7 -------
 common/public/property.te        |  1 -
 common/public/property_contexts  |  1 -
 qcom/private/recovery.te         |  6 ------
 qcom/sepolicy.mk                 |  3 ---
 10 files changed, 62 deletions(-)
 delete mode 100644 common/private/recovery.te
 delete mode 100644 common/private/system_app.te
 delete mode 100644 common/public/property.te
 delete mode 100644 common/public/property_contexts
 delete mode 100644 qcom/private/recovery.te

diff --git a/common/private/file_contexts b/common/private/file_contexts
index 71abfd1..83ceb58 100644
--- a/common/private/file_contexts
+++ b/common/private/file_contexts
@@ -8,9 +8,6 @@
 # OTA packages
 /data/lineageos_updates(/.*)?           u:object_r:ota_package_file:s0
 
-# Recovery
-/sbin/minivold                          u:object_r:vold_exec:s0
-
 # Sysinit
 /system/bin/sysinit                     u:object_r:sysinit_exec:s0
 
diff --git a/common/private/property.te b/common/private/property.te
index 1e6b220..8f1cc9e 100644
--- a/common/private/property.te
+++ b/common/private/property.te
@@ -1,3 +1,2 @@
 type adbtcp_prop, property_type;
-type recovery_prop, property_type;
 type userinit_prop, property_type;
diff --git a/common/private/property_contexts b/common/private/property_contexts
index ac89985..59832a3 100644
--- a/common/private/property_contexts
+++ b/common/private/property_contexts
@@ -1,4 +1,2 @@
 adb.network.port              u:object_r:adbtcp_prop:s0
-adb.secure                    u:object_r:recovery_prop:s0
 lineage.userinit.active       u:object_r:userinit_prop:s0
-recovery.perf.mode            u:object_r:recovery_prop:s0
diff --git a/common/private/recovery.te b/common/private/recovery.te
deleted file mode 100644
index d6f5295..0000000
--- a/common/private/recovery.te
+++ /dev/null
@@ -1,35 +0,0 @@
-recovery_only(`
-
-# XXX
-##allow init rootfs:file write;
-get_prop(recovery, sf_lcd_density_prop)
-r_dir_file(recovery, pstorefs)
-unix_socket_connect(recovery, vold, vold)
-
-# Secure adb (setup_adbd)
-allow adbd adb_keys_file:dir search;
-r_dir_file(recovery, adb_keys_file)
-set_prop(recovery, shell_prop)
-
-# Manage fstab and /adb_keys
-allow recovery rootfs:file create_file_perms;
-allow recovery rootfs:file link;
-allow recovery rootfs:dir { write create rmdir add_name remove_name };
-
-# Sideload cache
-allow recovery proc_meminfo:file { getattr open read };
-
-# Read storage files and directories
-allow recovery tmpfs:dir mounton;
-r_dir_file(recovery, media_rw_data_file)
-r_dir_file(recovery, sdcard_type)
-r_dir_file(recovery, vfat)
-
-# Control properties
-set_prop(recovery, ffs_prop)
-set_prop(recovery, recovery_prop)
-
-# Set system properties for various things
-set_prop(recovery, system_prop)
-
-')
diff --git a/common/private/system_app.te b/common/private/system_app.te
deleted file mode 100644
index 1b41d9a..0000000
--- a/common/private/system_app.te
+++ /dev/null
@@ -1,3 +0,0 @@
-# For the updaters
-allow system_app cache_recovery_file:dir {add_name rw_file_perms};
-allow system_app cache_recovery_file:file {create rw_file_perms};
diff --git a/common/private/vold.te b/common/private/vold.te
index a7fb84f..0ac6e81 100644
--- a/common/private/vold.te
+++ b/common/private/vold.te
@@ -1,5 +1,3 @@
-domain_trans(init, rootfs, vold)
-
 # NTFS-3g wants to drop permission
 allow vold self:capability { setgid setuid };
 
@@ -7,8 +5,3 @@ allow vold self:capability { setgid setuid };
 allow vold mkfs_exec:file { execute read open getattr execute_no_trans };
 allow vold mnt_media_rw_stub_file:dir r_dir_perms;
 allow vold storage_stub_file:dir { rw_file_perms search add_name };
-
-# Vold can also run as minivold in the rootfs
-recovery_only(`
-  allow vold rootfs:dir { add_name write };
-')
diff --git a/common/public/property.te b/common/public/property.te
deleted file mode 100644
index 81735e0..0000000
--- a/common/public/property.te
+++ /dev/null
@@ -1 +0,0 @@
-type sf_lcd_density_prop, property_type;
diff --git a/common/public/property_contexts b/common/public/property_contexts
deleted file mode 100644
index 36f5f3b..0000000
--- a/common/public/property_contexts
+++ /dev/null
@@ -1 +0,0 @@
-ro.sf.lcd_density             u:object_r:sf_lcd_density_prop:s0
diff --git a/qcom/private/recovery.te b/qcom/private/recovery.te
deleted file mode 100644
index dbdb205..0000000
--- a/qcom/private/recovery.te
+++ /dev/null
@@ -1,6 +0,0 @@
-recovery_only(`
-
-allow recovery sysfs_graphics:dir search;
-allow recovery sysfs_graphics:file { getattr open read write };
-
-')
diff --git a/qcom/sepolicy.mk b/qcom/sepolicy.mk
index c65816c..dcc5eab 100644
--- a/qcom/sepolicy.mk
+++ b/qcom/sepolicy.mk
@@ -3,9 +3,6 @@
 # that inherit from Lineage
 #
 
-BOARD_PLAT_PRIVATE_SEPOLICY_DIR += \
-    device/lineage/sepolicy/qcom/private
-
 BOARD_SEPOLICY_DIRS += \
     device/lineage/sepolicy/qcom/common \
     device/lineage/sepolicy/qcom/$(TARGET_BOARD_PLATFORM)
-- 
2.16.2

